extension/wallet/src/lib.rs

```diff
@@ -265,6 +265,35 @@ pub async fn sign_loan(wallet_name: String) -> Result<JsValue, JsValue> {
     Ok(loan_tx)
 }
 
+/// Create a json object which can be used to backup the loan details to the transaction
+///
+/// TODO: Pass in transaction id instead of transaction
+/// For now, we pass in the whole transaction just to access it's transaction id.
+/// This is because `sign_loan` returns the whole transaction and we have no means to deserialize
+/// the transaction from within TS.
+#[wasm_bindgen]
+pub async fn create_loan_backup(
+    wallet_name: String,
+    transaction: JsValue,
+) -> Result<JsValue, JsValue> {
+    log::debug!("Received tx: {:?}", transaction);
+    let transaction: Transaction = map_err_from_anyhow!(transaction.into_serde())?;
+    let backup_details = map_err_from_anyhow!(
+        wallet::create_loan_backup(wallet_name, &LOADED_WALLET, transaction.inner.txid()).await
+    )?;
+    let backup_details = map_err_from_anyhow!(JsValue::from_serde(&backup_details))?;
+    Ok(backup_details)
+}
+
+/// Load a loan backup and store it in localStorage.
+#[wasm_bindgen]
+pub async fn load_loan_backup(backup_details: JsValue) -> Result<(), JsValue> {
+    let backup_details: BackupDetails = map_err_from_anyhow!(backup_details.into_serde())?;
+    map_err_from_anyhow!(wallet::load_loan_backup(backup_details))?;
+
+    Ok(())
+}
```

Comment:
I wonder if it is a good idea to call things directly from TS code instead. 
Are you not afraid that the TS code and rust code diverge too much? 

For example, the loan data is currently stored in localstorage under the key `loan_state`. If this ever to changes in the rust code we might run in the danger that it's not changed in the TS code.

Reply:
I looked into it: in this particular case it's unfortunately a bigger task : 

The issue here is that we need to know the transaction ID to load this information from the localstorage. All we have available is the whole transaction. We'd need to change quite a few things for this. It's worth it and should be done but since it's working atm I recorded an issue for this. #238